src/controllers/api.js

```
@@ -10,10 +11,43 @@ const questionsGET = async (req, res, next) => {
     }
 }
 
+const quizzesGET = async (req, res, next) => {
+    try {
+        const result = await apiService.quizzesGET()
+
+        return res.status(200).json( { msg: result } )
+    } catch (error) {
+        next(error)
+    }
+}
+
+const tokensGET = async (req, res, next) => {
+    try {
+        const { token } = req.cookies || {}
+        const { email } = await jwt.decode(token) || {}
+
+        const result = await apiService.tokensGET(email)
+        console.log(result);
+
+        return res.status(200).json( { msg: result } )
+    } catch (error) {
+        next(error)
+    }
+}
+
 const personaPOST = async (req, res, next) => {
     try {
         const result = await apiService.personaPOST(req.body)
 
+        const { token } = req.cookies || {}
+        const { email } = await jwt.decode(token) || {}
+
+        try {
+            await apiService.decrementApiTokens(email);
+        } catch (err) {
```

Please throw the error in the service. Double catch is not good because this is not testable. Here the error is determined by the controller instead of the service. Therefore, there is no return/throw to be tested for the service.